azurerm/internal/services/domainservices/active_directory_domain_service_data_source.go

```diff
@@ -0,0 +1,338 @@
+package domainservices
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/domainservices/mgmt/2020-01-01/aad"
+
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tags"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/pluginsdk"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func dataSourceActiveDirectoryDomainService() *pluginsdk.Resource {
+	return &pluginsdk.Resource{
+		Read: dataSourceActiveDirectoryDomainServiceRead,
+
+		Timeouts: &pluginsdk.ResourceTimeout{
+			Read: pluginsdk.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: map[string]*pluginsdk.Schema{
+			"name": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringIsNotWhiteSpace,
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupNameForDataSource(),
+
+			"deployment_id": {
+				Type:     pluginsdk.TypeString,
+				Computed: true,
+			},
+
+			"domain_configuration_type": {
+				Type:     pluginsdk.TypeString,
+				Computed: true,
+			},
+
+			"domain_name": {
+				Type:     pluginsdk.TypeString,
+				Computed: true,
+			},
+
+			"filtered_sync_enabled": {
+				Type:     pluginsdk.TypeBool,
+				Computed: true,
+			},
+
+			"location": azure.SchemaLocationForDataSource(),
+
+			"notifications": {
+				Type:     pluginsdk.TypeList,
+				Computed: true,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"additional_recipients": {
+							Type:     pluginsdk.TypeList,
+							Computed: true,
+							Elem: &pluginsdk.Schema{
+								Type: pluginsdk.TypeString,
+							},
+						},
+						"notify_dc_admins": {
+							Type:     pluginsdk.TypeBool,
+							Computed: true,
+						},
+						"notify_global_admins": {
+							Type:     pluginsdk.TypeBool,
+							Computed: true,
+						},
+					},
+				},
+			},
+
+			"replica_sets": {
+				Type:     pluginsdk.TypeList,
+				Computed: true,
+				Elem: &pluginsdk.Resource{
+					Schema: dataSourceActiveDirectoryDomainServiceReplicaSetSchema(),
+				},
+			},
+
+			"resource_forest": {
+				Type:     pluginsdk.TypeList,
+				Computed: true,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"resource_forest": {
+							Type:     pluginsdk.TypeString,
+							Computed: true,
+						},
+
+						"forest_trust": {
```

Comment:
forest is in the block name?
```suggestion
						"trust": {
```

Reply:
Good call, have renamed those child attributes